lib/services/plugins-service.ts

```diff
@@ -43,14 +59,54 @@ export class PluginsService implements IPluginsService {
 
 	public remove(pluginName: string): IFuture<void> {
 		return (() => {
+			let isUninstallCommandExecuted = false;
+
+			let executeUninstallCommand = () => {
+				return (() => {
+					if(!isUninstallCommandExecuted) {
+						this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+						isUninstallCommandExecuted = true;
+					}
+				}).future<void>()();
+			};
+
 			let removePluginNativeCodeAction = (modulesDestinationPath: string, platform: string, platformData: IPlatformData) => {
-				let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
-				pluginData.isPlugin = true;
-				return platformData.platformProjectService.removePluginNativeCode(pluginData);
+				return (() => {
+					let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
+
+					platformData.platformProjectService.removePluginNativeCode(pluginData).wait();
+
+					// Remove the plugin and call merge for another plugins that have configuration file
+					let pluginConfigurationFilePath = this.getPluginConfigurationFilePath(pluginData, platformData);
+					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						let tnsModulesDestinationPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, constants.TNS_MODULES_FOLDER_NAME);
+						let nodeModules = this.$broccoliBuilder.getChangedNodeModules(tnsModulesDestinationPath, platform).wait();
+						let plugins = _.map(_.keys(nodeModules), nodeModule => this.convertToPluginData(this.getNodeModuleData(pluginName).wait()));
```

Comment:
This is only a suggestion and not a merge stopper:

```
_.(nodeModules)
.map(nodeModule => this.getNodeModuleData(pluginName).wait())
.map(nodeModuleData => this.convertToPluginData(nodeModuleData))
.filter(data => data.isPlugin && this.$fs.exists(this.getPluginConfigurationFilePath(data, platformData)).wait())
.forEach((data, index) => {
	executeUninstallCommand().wait();

	if(index === 0) {
		this.initializeConfigurationFileFromCache(platformData).wait();
	}

	if(data.name !== pluginName) {
		this.merge(data, platformData).wait();
	}
})
.value();
```

I find this code to be more linear but the original is correct and I do not insist on changing it.